apps/kdf.c

```diff
@@ -100,24 +86,57 @@ int kdf_main(int argc, char **argv)
         goto opthelp;
     }
 
-    id = OBJ_sn2nid(argv[0]);
-    if (id == NID_undef) {
+    if ((kdf = EVP_KDF_fetch(NULL, argv[0], NULL)) == NULL) {
         BIO_printf(bio_err, "Invalid KDF name %s\n", argv[0]);
         goto opthelp;
     }
 
-    ctx = EVP_KDF_CTX_new_id(id);
+    ctx = EVP_KDF_CTX_new(kdf);
     if (ctx == NULL)
         goto err;
 
     if (opts != NULL) {
-        for (i = 0; i < sk_OPENSSL_STRING_num(opts); i++) {
+        def = EVP_KDF_CTX_settable_params(kdf);
+        n = sk_OPENSSL_STRING_num(opts);
```

Comment:
You should look at https://github.com/openssl/openssl/blob/master/apps/include/apps.h#L267

Reply:
Good point, another simplification.
Done now.

Reply:
Or not....CI's

Reply:
Fixed now, case sensitivity issue over _N_ vs _n_.